pkg/liveshare/client.go

```diff
@@ -112,15 +119,52 @@ func Connect(ctx context.Context, opts Options) (*Session, error) {
 	s := &Session{
 		ssh:             ssh,
 		rpc:             rpc,
+		grpc:            grpc.NewClient(),
```

Comment:
It feels weird to me to have the grpc connection inside the LS connection - it feels like rather than have LS be aware of the existence of grpc it should be the caller decision and responsibility to set up grpc after setting up LS

In the back of my mind I'm thinking of the migrations away from LS to tunnels, and so when that happens we'll need a whole new `TunnelSession` type class which, with this current design, would also need to internally set up grpc right?

In the caller code, could we have it be something like this:

```go
session, _ := liveshare.Connect(...)
grpc, _ := grpc.Connect(session, ...)
```

This also removes the need to thin wrappers like `StartJupyterServer` which can instead just invoke `grpc.StartJupyterServer` directly rather than the redirection through `session`

Reply:
I am not opposed to this 🤔 

I guess it depends on what the design looks like after we move past simple operations like starting the jupyter server and more like port forwarding and session heartbeats. At that point, it feels like we need a new concept to group around that's not just grpc. With this approach, our commands will need to start knowing about grpc (perhaps that's a good thing)

However, I think this current approach is an excellent first step at replacing the current operations with grpc while keeping the changeset small. At some point, this will all require another refactor.

Reply:
Agree that we don't necessarily need to take that on in this PR, but it would be nice to at least have consensus on the overall design goal that these PRs can then move towards

Reply:
For sure! Would love to know what's coming next, as it would make the reviews easier.

Reply:
Went ahead and moved the gRPC client out of the LS session and into the individual command call like how Caleb proposed. That will definitely make it easier when it comes time to make the connection stack switch in the future.